transports/jms/src/main/java/org/mule/transport/jms/JmsConnector.java

```diff
@@ -432,7 +432,7 @@ protected Connection createConnection() throws MuleException, JMSException
 
         Connection connection;
 
-        if (username != null)
+        if (!cacheJmsSessions && username != null)
```

Comment:
what happens if I want to use session caching and still have authentication? Can't I cache the sessions for this user?